### ISSUE TITLE
_mkdir wierdness

### ISSUE THREAD

**Issue by rtoijala** [scp-fs2open/fs2open.github.com#855]
I'm having trouble creating a new pilot on Linux, and I've traced the problem to a bug in `code/windows_stubs/stubs.cpp` in the `_mkdir` function.

If passed a path like `/foo/bar/baz`, `_mkdir` succesfully creates `/foo` and `/foo/bar/` but does not attempt to create `/foo/bar/baz` since at that point `c` is `NULL` and the if clause is not executed.
This in itself is a bug that can be fixed by appending a slash to the given path if it does not have one.

The reason I am writing this issue instead of creating a pull request is that (as far as I can see), `_mkdir` should not create directories recursively at all, since (as far as I know) it is not recursive on Windows. Therefore calling it with a full path now has different effects on Windows and Linux (and Mac?). Also, some places (notably `cfile.cpp:mkdir_recursive` and `cfileextractor.cpp:extract_all_files`) that call `_mkdir` already loop over the path parts, duplicating what `_mkdir` is already doing. Places that do not loop are `freespace.cpp:actual_main` when creating the path returned by `os_get_config_path()` and `gropengl.cpp:gr_opengl_print_screen` when creating the screenshots folder.

I'm not sure how to proceed, so here I am asking.

It is also interesting to note that `_mkdir` was last edited in 2006, so I am at a loss as to how this has ever worked...

**Comment by rtoijala** [comment]
Just to make the issue clearer, here is the log output when trying to create a new pilot file.
Notice that the `players` directory does not exist, since this is my first pilot.
I've added logging to `windows_stubs.cpp:_mkdir` so it shows all calls to mkdir.
Notice also that _mkdir does not attempt to create the `players` directory.
```
CFILE: Creating new directory '/home/risto/.local/share/HardLightProductions/FreeSpaceOpen/data/players/'
mkdir("/home")
mkdir("/home")
mkdir("/home/risto")
mkdir("/home")
mkdir("/home/risto")
mkdir("/home/risto/.local")
mkdir("/home")
mkdir("/home/risto")
mkdir("/home/risto/.local")
mkdir("/home/risto/.local/share")
mkdir("/home")
mkdir("/home/risto")
mkdir("/home/risto/.local")
mkdir("/home/risto/.local/share")
mkdir("/home/risto/.local/share/HardLightProductions")
mkdir("/home")
mkdir("/home/risto")
mkdir("/home/risto/.local")
mkdir("/home/risto/.local/share")
mkdir("/home/risto/.local/share/HardLightProductions")
mkdir("/home/risto/.local/share/HardLightProductions/FreeSpaceOpen")
mkdir("/home")
mkdir("/home/risto")
mkdir("/home/risto/.local")
mkdir("/home/risto/.local/share")
mkdir("/home/risto/.local/share/HardLightProductions")
mkdir("/home/risto/.local/share/HardLightProductions/FreeSpaceOpen")
mkdir("/home/risto/.local/share/HardLightProductions/FreeSpaceOpen/data")
PLR => Unable to open 'haha.plr' for saving!
```

**Comment by asarium** [comment]
Both existing calls to `_mkdir` are either not needed or only need to create one path element. The call in freespace.cpp is not needed since the path returned by `os_get_config_path()` is guaranteed to already exist. The same can be applied to `gr_opengl_print_screen` where it only needs to create one path element since the config path already exists.

If you want, you can make `_mkdir` a normal alias for `mkdir` or the other way around and submit a pull request for that.

**Comment by rtoijala** [comment]
Fixed by #857.
